test/common/formatter/substitution_formatter_test.cc

```diff
@@ -1224,6 +1224,21 @@ TEST(SubstitutionFormatterTest, responseTrailerFormatter) {
   }
 }
 
+TEST(SubstitutionFormatterTest, escapeFormatter) {
+  EscapeFormatter formatter;
+  Http::TestRequestHeaderMapImpl request_headers{{":method", "GET"}, {":path", "/"}};
+  Http::TestResponseHeaderMapImpl response_headers;
+  Http::TestResponseTrailerMapImpl response_trailers;
+  StreamInfo::MockStreamInfo stream_info;
+  std::string body;
+
+  EXPECT_EQ("%", formatter.format(request_headers, response_headers, response_trailers, stream_info,
+                                  body));
+  EXPECT_THAT(formatter.formatValue(request_headers, response_headers, response_trailers,
```

Comment:
This tests that EscapeFormatter works correctly, but not the EscapeFormatter is actually used for a given format string. We'll need to add another test that actually parses a format string containing `%%` -- the one you gave in your earlier comment is probably a good test case -- and show that it it's rendered as expected.